midend/parserUnroll.cpp

```diff
@@ -147,19 +225,19 @@ class ParserStateRewriter : public Transform {
             BUG_CHECK(l->is<SymbolicArray>(), "%1%: expected an array", l);
             auto array = l->to<SymbolicArray>();
             unsigned idx = 0;
-            for (size_t i = 0; i < array->size; i++) {
-                auto* v = array->get(expression, i);
-                if (v->hasUninitializedParts())
-                    break;
-                else
-                    idx = i;
+            unsigned offset = 0;
```

Comment:
can you explain what is happening here?

Reply:
if it is a next operator then we need to make offset = 1, otherwise it should be 0. I did some refactoring there.